pkg/scheduler/invoker/invoker.go

```diff
@@ -36,7 +40,9 @@ func (p *Params) newRemoteTask(callbackURL string) *reconciler.Task {
 
 func (p *Params) newTask() *reconciler.Task {
 	version := p.ClusterState.Configuration.KymaVersion
-	if p.ComponentToReconcile.Version != "" {
+	if p.ComponentToReconcile.URL != "" && p.ComponentToReconcile.Version == "" {
+		version = defaultBranch
+	} else if p.ComponentToReconcile.Version != "" {
 		version = p.ComponentToReconcile.Version
```

Comment:
We should actually not set a default here (at least my opinion): we cannot know which naming convention is used in an external GIT repository (maybe they use `master` as their mainstream branch etc.).

I would recommend to make the version mandatory for external components hosted in a GIT repository and not to use any default here.

In this case: the version should be provided as part of the component configuration.

Reply:
Please set the version only for GIT urls.